### PR TITLE
Added mixed and velocity-dependent mixed BCs, CORs, and tests. 

### PR DESCRIPTION
**Pull request by Agar1204**
I built the following features: 

BC=3 (Mixed): Added a mixed absorbing/reflecting boundary condition. A fraction mixed_BC_weight of the macroparticle is reflected (position mirrored, velocity flipped); the remainder is absorbed. This is implemented by scaling the macroparticle's charge q and mass m by mixed_BC_weight on each wall hit. mixed_BC_weight=1.0 recovers fully reflective (BC=1); mixed_BC_weight=0.0 recovers fully absorbing (BC=2).

BC=4 (Velocity-dependent mixed): Added a velocity-dependent mixed BC where the reflected fraction is clamp(1 - |vx| / max_vx, 0, 1), with max_vx = max(|vx|) computed dynamically across all particles each timestep. Slow particles reflect more; fast particles are absorbed more. Only q and m are scaled — q_m is left unchanged so the particle's charge-to-mass ratio stays correct for field calculations.

Coefficient of Restitution (COR): Added COR_left and COR_right parameters (defaulting to 1.0) controlling energy loss on wall reflection: vx_after = -COR * vx_before. Applies to all reflecting BCs (BC=1, 3, 4) since COR is a wall property. Only the normal component vx is scaled; vy and vz are unchanged. Validated to [0, 1].

Diagnostics fix: _diagnostics.py now uses time-varying masses (masses_over_time) when computing total kinetic energy, rather than fixed initial masses. This required adding ms to the per-step output tuple in _algorithms.py so masses are tracked across the simulation. This was necessary for mixed BCs where macroparticle mass decreases on each wall hit.

Refactor: Replaced nested jnp.where chains in _boundary_conditions.py with jnp.select for readability and extensibility to new BC types.

Tests: Added tests to tests/test_boundary_conditions.py covering: BC=3 single-particle and batch behavior (charge/mass scaling, position reflection, velocity flip); BC=4 velocity-dependent weight (slow vs. fast particle behavior, q_m invariance); COR scaling for single particles and batches across BC=1, 3, and 4; and edge cases (particle inside domain unchanged, COR=1 recovers elastic reflection).

### CODE REVIEW

**Comment by rogeriojorge** [comment]
Thank you for the work!
Can you add an example showcasing the new boundary conditions? Also, can you document in the README what the new parameters are, such as COR?

File: jaxincell/_algorithms.py [hunk 1]
```diff
@@ -90,8 +90,8 @@ def interpolate_fields(x_n):
     charge_density = calculate_charge_density(positions, qs, dx, grid, particle_BC_left, particle_BC_right,
                                               filter_passes=fpasses, filter_alpha=falpha, filter_strides=fstrides,
                                               field_BC_left=field_BC_left, field_BC_right=field_BC_right)
-    step_data = (positions, velocities, E_field, B_field, J, charge_density)
-    
+    step_data = (positions, velocities, ms, E_field, B_field, J, charge_density)
+
     return carry, step_data
```
[the remark this answers is not in the document]
Comment (Agar1204) [reply jaxincell/_algorithms.py:95]: In simulation.py, we set mixed_BC_weight, COR_left, and COR_right parameters to 1.0 as default if the user's TOML input file doesn't contain these parameters. These default parameters are then passed into Boris_step and CN_step, meaning we don't need the default values added here.

**Comment by Agar1204** [comment]
@rogeriojorge 

Added two examples showcasing the new boundary conditions, implemented some of the relevant suggestions by copilot, and updated the README to include documentation for any new parameters. Let me know if anything else needs to be changed!
